### ISSUE TITLE
Feature request: Percentile across dimensions

### ISSUE THREAD

**Issue by holtc** [josdejong/mathjs#1552]
Is there a way to get the xth percentile of data across a matrix dimension?  Similar to how `mean` returns the mean vector across a given dimension?  Similar to [numpy's percentile method](https://docs.scipy.org/doc/numpy/reference/generated/numpy.percentile.html), where we could pass in an axis

**Comment by josdejong** [comment]
mathjs currently doesn't have a `percentile` function. Would be nice to implement this. And when implementing, it makes sense to implement support for specifying a dimension too, like other matrix function like `mean` do. Thanks for your suggestion Chris.

Anyone interested in implementing `percentile` for mathjs?

**Comment by holtc** [comment]
For now, I have a naive implementation that involves transposing the matrix to get the result I desire:

```
const transposed = math.transpose(matrix)
const percentileVector = []

transposed.forEach(row => {
  percentileVector.push(math.quantileSeq(row, percentile))
})
```

**Comment by dvd101x** [comment]
Hi Jos, I'm interested in this topic, can I attempt a PR for this?

**Comment by josdejong** [comment]
Thanks David. Yes a PR would be very welcome!

**Comment by gwhitney** [comment]
Implemented in #3002, closing.
